services/s3management/accesskeys.go

```diff
@@ -0,0 +1,165 @@
+	Description     types.String `tfsdk:"description"`
+	ID              types.String `tfsdk:"id"`
+}
+
+func (c *Client) GetAccessKey(ctx context.Context, accessKeyId string) (s3management.AccessKey, *s3management.APIResponse, error) {
+	accessKey, apiResponse, err := c.client.AccesskeysApi.AccesskeysFindById(ctx, accessKeyId).Execute()
+	apiResponse.LogInfo()
+	return accessKey, apiResponse, err
+}
+
+func (c *Client) ListAccessKeys(ctx context.Context) (s3management.AccessKeyList, *s3management.APIResponse, error) {
+	accessKeys, apiResponse, err := c.client.AccesskeysApi.AccesskeysGet(ctx).Execute()
+	apiResponse.LogInfo()
+	return accessKeys, apiResponse, err
+}
+
+func (c *Client) CreateAccessKey(ctx context.Context, accessKey s3management.AccessKeyCreate, timeout time.Duration) (s3management.AccessKey, *s3management.APIResponse, error) {
+	accessKeyResponse, apiResponse, err := c.client.AccesskeysApi.AccesskeysPost(ctx).AccessKeyCreate(accessKey).Execute()
+	apiResponse.LogInfo()
+	if err == nil && accessKeyResponse.Id != nil {
+		err = backoff.Retry(func() error {
+			return c.accessKeyAvailableCheck(ctx, *accessKeyResponse.Id)
+		}, backoff.NewExponentialBackOff(backoff.WithMaxElapsedTime(timeout)))
+		if err != nil {
+			return accessKeyResponse, apiResponse, fmt.Errorf("failed to wait for accessKey available: %w", err)
+		}
+	}
+
+	return accessKeyResponse, apiResponse, err
+}
+
+func (c *Client) UpdateAccessKey(ctx context.Context, accessKeyId string, accessKey s3management.AccessKeyEnsure, timeout time.Duration) (s3management.AccessKey, *s3management.APIResponse, error) {
+	accessKeyResponse, apiResponse, err := c.client.AccesskeysApi.AccesskeysPut(ctx, accessKeyId).AccessKeyEnsure(accessKey).Execute()
+	apiResponse.LogInfo()
+	if err == nil {
+		err = backoff.Retry(func() error {
+			return c.accessKeyAvailableCheck(ctx, accessKeyId)
+		}, backoff.NewExponentialBackOff(backoff.WithMaxElapsedTime(timeout)))
+		if err != nil {
+			return accessKeyResponse, apiResponse, fmt.Errorf("failed to wait for accessKey available: %w", err)
+		}
+	}
+
+	return accessKeyResponse, apiResponse, err
+}
+
+func (c *Client) DeleteAccessKey(ctx context.Context, accessKeyId string, timeout time.Duration) (*s3management.APIResponse, error) {
+	apiResponse, err := c.client.AccesskeysApi.AccesskeysDelete(ctx, accessKeyId).Execute()
+	apiResponse.LogInfo()
+
+	if err == nil {
+		err = backoff.Retry(func() error {
+			return c.accessKeyDeletedCheck(ctx, accessKeyId)
+		}, backoff.NewExponentialBackOff(backoff.WithMaxElapsedTime(timeout)))
+		if err != nil {
+			return apiResponse, fmt.Errorf("failed to wait for accessKey available: %w", err)
+		}
+	}
+
+	return apiResponse, err
+}
+
+func SetAccessKeyPropertiesToPlan(plan *AccesskeyResourceModel, accessKey s3management.AccessKey) {
+
+	if accessKey.Properties != nil {
+		if accessKey.Properties.AccessKey != nil {
+			plan.AccessKey = basetypes.NewStringPointerValue(accessKey.Properties.AccessKey)
```

Comment:
why are you checking for nil if you use PointerValue?

Reply:
I do not want to overwrite an existing value with nil, it happens with secret key which is only returned on create requests

Reply:
but this happens only for secret key, right? Maybe we can check only for that.

Reply:
also happens for canonicalUserId, seemed safer to just check them all

Reply:
Maybe a comment explaining this would be enough

Reply:
sure, I will add one